packages/rainbowkit/src/wallets/connectorsForWallets.ts

```diff
@@ -1,32 +1,29 @@
 import { Connector } from 'wagmi';
 import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
+import { isMobile } from '../utils/isMobile';
 import { omitUndefinedValues } from '../utils/omitUndefinedValues';
 import { ConnectorArgs, WalletInstance, WalletList } from './Wallet';
 
 export const connectorsForWallets = (walletList: WalletList) => {
+  let index = -1;
+
   return function (connectorArgs: ConnectorArgs) {
     const connectors: Connector[] = [];
 
     walletList.forEach(({ groupName, wallets }) => {
       wallets.forEach(({ createConnector, ...walletMeta }) => {
+        index++;
+
         const { connector, ...connectionMethods } = omitUndefinedValues(
           createConnector(connectorArgs)
         );
 
-        // @ts-expect-error
-        if (connector._wallet) {
-          throw new Error(
-            `Can't connect wallet "${walletMeta.name}" to connector "${
-              connector.name ?? connector.id
-            }" as it's already connected to wallet "${
-              // @ts-expect-error
-              connector._wallet.name
-            }". Each wallet must have its own connector instance.`
-          );
-        }
-
         let walletConnectModalConnector: Connector | undefined;
-        if (walletMeta.id === 'walletConnect' && connectionMethods.qrCode) {
+        if (
+          walletMeta.id === 'walletConnect' &&
+          connectionMethods.qrCode &&
+          !isMobile()
```

Comment:
This is an added optimisation. If the user is on mobile, this connector was never used, so we can avoid another websocket connection.